src/output.cpp

```diff
@@ -1053,25 +1103,24 @@ void full_screen_popup(const char *mes, ...)
 int draw_item_info(const int iLeft, const int iWidth, const int iTop, const int iHeight,
                    const std::string sItemName,
                    std::vector<iteminfo> &vItemDisplay, std::vector<iteminfo> &vItemCompare,
-                   const int selected, const bool without_getch, const bool without_border)
+                   int &selected, const bool without_getch, const bool without_border, const bool handle_scrolling)
 {
     WINDOW *win = newwin(iHeight, iWidth, iTop + VIEW_OFFSET_Y, iLeft + VIEW_OFFSET_X);
 
     const auto result = draw_item_info(win, sItemName, vItemDisplay, vItemCompare,
-                          selected, without_getch, without_border);
+                          selected, without_getch, without_border, handle_scrolling);
     delwin( win );
     return result;
 }
 
 int draw_item_info(WINDOW *win, const std::string sItemName,
                    std::vector<iteminfo> &vItemDisplay, std::vector<iteminfo> &vItemCompare,
-                   const int selected, const bool without_getch, const bool without_border)
+                   int &selected, const bool without_getch, const bool without_border, const bool handle_scrolling)
 {
+    std::ostringstream buffer;
     int line_num = 1;
     if (sItemName != "") {
-        const int iOffset = (without_border) ? 0 : 2;
-        trim_and_print(win, line_num, iOffset, getmaxx(win) - iOffset, c_white, "%s", sItemName.c_str());
-        line_num = 3;
+        buffer << sItemName.c_str() << "\n \n";
```

Comment:
Pointless use of `c_str`, you can print a `std::string` directly. Btw. why is there a space between the two newlines? (If it's required for some reason, that should probably be commented.)